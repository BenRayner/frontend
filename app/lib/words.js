// @flow

const VOWELS = 'aeiou';

export const indefiniteArticleFor = (word: string): string => {
  const first = ('' + word).toLowerCase()[0];

  return VOWELS.indexOf(first) !== -1
    ? 'an'
    : 'a';
};
