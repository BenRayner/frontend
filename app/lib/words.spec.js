/* global describe, it, expect */
import { indefiniteArticleFor } from './words';

const TEST_WORDS = [
  'pipeline',
  'organization',
  'request',
  'dog',
  'cat',
  'ant',
  'clock'
];

describe('indefiniteArticleFor', () => {
  describe('correctly identifies indefinite articles', () => {
    TEST_WORDS.forEach((word) => {
      it(`for “${word}”`, () => {
        const articleForWord = indefiniteArticleFor(word);
        const articleForUppercaseWord = indefiniteArticleFor(word.toUpperCase());

        expect(articleForWord).toMatchSnapshot();
        expect(articleForUppercaseWord).toMatchSnapshot();
        expect(articleForWord).toEqual(articleForUppercaseWord);
      });
    });
  });
});
