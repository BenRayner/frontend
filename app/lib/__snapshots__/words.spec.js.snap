// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`indefiniteArticleFor correctly identifies indefinite articles for “ant” 1`] = `"an"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “ant” 2`] = `"an"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “cat” 1`] = `"a"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “cat” 2`] = `"a"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “clock” 1`] = `"a"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “clock” 2`] = `"a"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “dog” 1`] = `"a"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “dog” 2`] = `"a"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “organization” 1`] = `"an"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “organization” 2`] = `"an"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “pipeline” 1`] = `"a"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “pipeline” 2`] = `"a"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “request” 1`] = `"a"`;

exports[`indefiniteArticleFor correctly identifies indefinite articles for “request” 2`] = `"a"`;
